code/scheme/guile-r7rs/run

```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+GUILE=$(which guile)
+if [ "$GUILE" = "" ]; then
+    case $(uname) in
+        Darwin)
+            echo "Install guile, see e.g. https://formulae.brew.sh/formula/guile"
+            ;;
+        *)
+            echo "Install guile on your system, see https://www.gnu.org/software/guile/download/"
+            ;;
+    esac
+    exit 1
```

Comment:
Instead of installing Guile locally here, just provide pointers to installation via pacakge managers, which is the recommended installation method. I tried to build Guile locally on my Apple Silicon Mac, but `./configure` failed with an error.

Reply:
This is why I gave up on Guile: I cannot install Guile on my M1 Mac because I am not using Homebrew nor MacPorts but rather build what I need from source, or download a prebuilt binary for macOS which Guile does not provide. Also on WSL 1 (Ubuntu) building autoconf from source also fails and Guile will not build with its default autoconf version 2.69. I no longer install software that require sudo or want to be in my system level /usr/ or /opt/ directories which the GNU tools seem to require. If you can make a video or some way of proving that this PR will run BBHW successfully, then I will approve it. Thanks.

Reply:
GNU tools should not require to be in system level directories; you can usually configure them to be wherever with --prefix. Releases should typically have run autoconf already and include the configure script it generated.

Reply:
Guile has dependencies to other GNU libraries, which makes building it quite involved. However, I tried by configuring guile with `--prefix` and then pulling the release of the first dependency, `libunistring`, where guile configure stopped and then configuring and building it with the same prefix directory. Unfortunately `libunistring` `make install` fails on my machine. The Homebrew formula for building Guile doesn't look simple either: https://github.com/Homebrew/homebrew-core/blob/b302ac97430030e01f0958eb512b8de4fecc55ca/Formula/g/guile.rb#L44-L74.

Reply:
The culprit in installing `libunistring` locally was that the working directory I used had spaces in its absolute path. Seems like there is some path quoting issues in autoconf or in the `libunistring` build scripts.

I was able to finally build Guile locally in the `run` script, if not installed on the system, see commit e8b2b2b. Can you please test that it works also for you? I've tested only on macOS 14.5 with M1 CPU. Do note that the build will probably fail if the project working directory has space characters in its path.